libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1464,6 +1464,10 @@ void AP_InertialSensor::wait_for_sample(void)
  */
 bool AP_InertialSensor::get_delta_angle(uint8_t i, Vector3f &delta_angle) const
 {
+    if (i >= INS_MAX_INSTANCES) {
```

Comment:
should be get_gyro_count() not INS_MAX_INSTANCES I think

Reply:
That sounds wrong to me.

Reply:
This is a minimum change as it moves the i < INS_MAX_INSTANCES check from the EKF into AP_InertialSensor.

Reply:
Ah, never mind, it was checking get_accel_count. Are you sure that is correct? Why would get_accel_count and get_gyro_count be different? In the case where they are different, are the gyros and accels guaranteed to be assigned contiguous indices?